### PR TITLE
tests: improve test case descriptions of some interrupt APIs

### PR DESCRIPTION
**Pull request by enjiamai**
To add some test case descriptions for testing irq_lock(), irq_unlock(), irq_enable(), irq_disable(), includes descriptions, test step and pass criteria, etc.

Signed-off-by: Enjia Mai <enjiax.mai@intel.com>

### CODE REVIEW

File: tests/kernel/context/src/main.c [hunk 1]
```diff
@@ -451,15 +487,52 @@ static void test_kernel_interrupts(void)
...
+ * -# Do action to align to a tick boundary.
+ * -# Left shift 4 bit for the value of counts.
+ * -# Call irq_disable() and restore its return value to imask.
+ * -# Call z_tick_get_32() and store its return value to tick.
+ * -# Repeat counts of calling z_tick_get_32().
+ * -# Call z_tick_get_32() and store its return value to tick2.
+ * -# Call irq_enable() with parameter imask.
+ * -# Check if tick is equaled to tick2.
+ * -# Repeat counts of calling z_tick_get_32().
+ * -# Call z_tick_get_32() and store its return value to tick2.
```
Comment (LeiW000) [inline tests/kernel/context/src/main.c:522]: Same as line 465.

**Review by LeiW000 — CHANGES_REQUESTED** [review]

File: tests/kernel/context/src/main.c [hunk 2]
```diff
@@ -430,15 +430,51 @@ static void _test_kernel_interrupts(disable_int_func disable_int,
...
+ * -# Left shift 4 bits for the value of counts.
+ * -# Call irq_lock() and restore its return value to imask.
+ * -# Call z_tick_get_32() and store its return value to tick.
+ * -# Repeat counts of calling z_tick_get_32().
+ * -# Call z_tick_get_32() and store its return value to tick2.
+ * -# Call irq_unlock() with parameter imask.
+ * -# Check if tick is equaled to tick2.
+ * -# Repeat counts of calling z_tick_get_32().
+ * -# Call z_tick_get_32() and store its return value to tick2.
+ * -# Check if tick is NOT equaled to tick2.
```
[the remark this answers is not in the document]
Comment (enjiamai) [reply tests/kernel/context/src/main.c]: Got it fixed, thanks!

File: tests/kernel/context/src/main.c (hunk 1 above)
Comment (enjiamai) [reply tests/kernel/context/src/main.c:522]: Got it fixed, thanks!

**Review by LeiW000 — APPROVED** [review]

**Comment by LeiW000** [comment]
@mengxianglinx , please take look at it.

**Comment by LeiW000** [comment]
@mengxianglinx , please take look at the PR.

**Review by nashif — APPROVED** [review]

**Pull request merged**